akka-jackson/src/main/scala/akka/jackson/JacksonSerializer.scala

```diff
@@ -0,0 +1,200 @@
+  }
+
+  private val compressLargerThan: Long = conf.getBytes("compress-larger-than")
+
+  override def manifest(obj: AnyRef): String = {
+    val className = obj.getClass.getName
+    migrations.get(className) match {
+      case Some(transformer) ⇒ className + "#" + transformer.currentVersion
+      case None              ⇒ className
+    }
+  }
+
+  override def toBinary(obj: AnyRef): Array[Byte] = {
+    val startTime = if (isDebugEnabled) System.nanoTime else 0L
+    val bytes = objectMapper.writeValueAsBytes(obj)
+    val result =
+      if (bytes.length > compressLargerThan) compress(bytes)
+      else bytes
+
+    if (isDebugEnabled) {
+      val durationMicros = (System.nanoTime - startTime) / 1000
+      if (bytes.length == result.length)
+        log.debug(
+          "Serialization of [{}] took [{}] µs, size [{}] bytes",
+          obj.getClass.getName, durationMicros, result.length
+        )
+      else
+        log.debug(
+          "Serialization of [{}] took [{}] µs, compressed size [{}] bytes, uncompressed size [{}] bytes",
+          obj.getClass.getName, durationMicros, result.length, bytes.length
+        )
+    }
+
+    result
+  }
+
+  override def fromBinary(bytes: Array[Byte], manifest: String): AnyRef = {
+    val startTime = if (isDebugEnabled) System.nanoTime else 0L
+    val compressed = isGZipped(bytes)
+
+    val (fromVersion, manifestClassName) = parseManifest(manifest)
+
+    val migration = migrations.get(manifestClassName)
+
+    val className = migration match {
+      case Some(transformer) if fromVersion < transformer.currentVersion ⇒
+        transformer.transformClassName(fromVersion, manifestClassName)
+      case Some(transformer) if fromVersion > transformer.currentVersion ⇒
+        throw new IllegalStateException(s"Migration version ${transformer.currentVersion} is " +
+          s"behind version $fromVersion of deserialized type [$manifestClassName]")
+      case _ ⇒ manifestClassName
+    }
+
+    // FIXME Security concern: we would need a configured whitelist of class/package prefixes
+    // to prevent loading arbitary classes, i.e. app must at least configure its root package name
+
+    val clazz = system.dynamicAccess.getClassFor[AnyRef](className) match {
+      case Success(c) ⇒ c
+      case Failure(_) ⇒
+        throw new NotSerializableException(
+          s"Cannot find manifest class [$className] for serializer [${getClass.getName}]."
+        )
+    }
+
+    val decompressBytes = if (compressed) decompress(bytes) else bytes
+
+    if (isDebugEnabled) {
+      val durationMicros = (System.nanoTime - startTime) / 1000
+      if (bytes.length == decompressBytes.length)
+        log.debug(
+          "Deserialization of [{}] took [{}] µs, size [{}] bytes",
+          clazz.getName, durationMicros, decompressBytes.length
+        )
+      else
+        log.debug(
+          "Deserialization of [{}] took [{}] µs, compressed size [{}] bytes, uncompressed size [{}] bytes",
```

Comment:
Looks like this is only the decompression time that is logged here.
the whole block should be moved at the end of the method to measure total deserialization time.

Reply:
Right, thanks :)

Reply:
fixed